### ISSUE TITLE
Expose pendingRequests

### ISSUE THREAD

**Issue by KaelWD** [rollup/rollup#4294]
### Feature Use Case

I'm writing a plugin that needs to make modifications to a file based on other files in the project. In vite I'm using the `transform` hook with `server._pendingRequests` to check if I'm the only module left, but there's nothing like this exposed in rollup. 

### Feature Proposal

Something similar to vite: https://github.com/vitejs/vite/blob/b625a2cf3730fda94d934b23bcec0c859bfa0b24/packages/vite/src/node/server/transformRequest.ts#L49-L52

`ModuleLoader`:

```diff
+    public pendingRequests = new Map()
     async fetchModule({ id, meta, moduleSideEffects, syntheticNamedExports }, importer, isEntry) {
...
         this.graph.watchFiles[id] = true;
-        await this.addModuleSource(id, importer, module);
+        const promise = this.addModuleSource(id, importer, module)
+        this.pendingRequests.set(id, promise.then(() => module))
+        await promise;
+        this.pendingRequests.delete(id)
         const resolveStaticDependencyPromises = this.getResolveStaticDependencyPromises(module)
```

`PluginContext`:
```diff
    const context = {
+       pendingRequests: graph.moduleLoader.pendingRequests,
        addWatchFile(id) {
```

**Comment by lukastaegert** [comment]
While you do not have this exact API, you can already do something similar with a combination of `this.load` and `this.getModuleIds`.

The `id` of a module is added to `this.getModuleIds()` before its `load` hook is even run, so all "pending" modules can be found here. When you have the id, you can `await this.load(id)` to wait until it is loaded and get its module info.

**Comment by KaelWD** [comment]
That seems to work, thanks! I did still need a timeout too though because `this.load(id)` sometimes resolves instantly but `module.code` is still null, so I'd end up in an infinite loop.

**Comment by lukastaegert** [comment]
> sometimes resolves instantly but module.code is still null

That should not happen, would be interested if you could reproduce it.

**Comment by KaelWD** [comment]
Sure: https://github.com/KaelWD/rollup-load-null

**Comment by lukastaegert** [comment]
Thanks so much, there was actually a bug in the logic. Fix at #4296.
